Support for updating `.order` files in subfolders

### Azure DevOps Extensions

WIKI Updater

### Is your feature request related to a problem? Please describe.

As we are generating release notes for multiple builds/stages, we nest the related pages in our wiki: 
```
Release Notes
	\ Project 1
		v1
		v2
		...
	\ Project 2
		v1
		v2
		...
	\ Project 3
		v1
		v2
		...
```

When using the `updateOrderFile` option, only the `.order` file at the root of the wiki repo is updated, which does not influence the order of the pages in any of the subfolders.

### Describe the solution you'd like

A new option that controls the placement of new pages in the respective subfolder. In our case, that would result in:
```
Release Notes
	\ Project 1
		v2
		v1
		...
	\ Project 2
		v2
		v1
		...
	\ Project 3
		v2
		v1
		...
```


### Describe alternatives you've considered

The desired functionality can also be implemented with a custom powershell/bash script - but now you're forced to interact directly with a second git repository in your script, which isn't ideal.

### Additional context

_No response_

@rfennell 
Unfortunately, my first test of the new feature was unsuccessful :/

Log output:
```
Starting: WikiUpdaterTask
==============================================================================
Task         : Git based WIKI Single File Updater from Black Marble
Description  : A tools to update or create a file in a WIKI based in a Git repo 
Version      : 1.35.4
Author       : Richard Fennell
Help         : Version: 1.35.4. [More Information](https://github.com/rfennell/vNextBuild/wiki/Wiki-Updater-Task/)
==============================================================================
Variable: Repo [https://<org>@dev.azure.com/<org>/<project>/_git/<project>.wiki]
Variable: Filename [Release/test/myNewWikiPage.md]
Variable: Contents [undefined]
Variable: Commit Message [Update release notes for myNewWikiPage]
Variable: Git Username [9de84241-6340-6dfa-8158-98aefae37ada]
Variable: Git Email [<name>@<org>]
Variable: Use Agent Token [true]
Variable: Replace File [false]
Variable: Append to File [false]
Variable: Username [undefined]
Variable: Password [*****]
Variable: LocalPath [/home/vsts/work/1/s\repo]
Variable: Data Is File [true]
Variable: SourceFile [myNewWikiPage.md]
Variable: Tag Repo [false]
Variable: Tag [undefined]
Variable: Branch [wikiMaster]
Variable: InjectExtraHeader [false]
Variable: Retries [5]
Variable: trimLeadingSpecialChar [false]
Variable: fixLineFeeds [true]
Variable: fixSpaces [false]
Variable: insertLinefeed [false]
Variable: updateOrderFile [true]
Variable: prependEntryToOrderFile [true]
Variable: orderFilePath [Release/test]
Using OAUTH Agent Token, overriding username and password
The provided repo URL is https://<org>@dev.azure.com/<org>/<project>/_git/<project>.wiki
The protocol is https
The provided repo URL is https://<org>@dev.azure.com/<org>/<project>/_git/<project>.wiki
Removing leading http:// or https:// block
Removing leading username@ block
Trimmed the URL to dev.azure.com/<org>/<project>/_git/<project>.wiki
URL used https://buildagent:***@dev.azure.com/<org>/<project>/_git/<project>.wiki
Cleaned /home/vsts/work/1/s\repo
Cloned dev.azure.com/<org>/<project>/_git/<project>.wiki to /home/vsts/work/1/s\repo
Set GIT values in /home/vsts/work/1/s\repo
Creating the directory /home/vsts/work/1/s\repo/Release/test
Checking out the requested branch wikiMaster
Pull in case of post clone updates from other users
Working file name is myNewWikiPage.md
Prepending to the myNewWikiPage.md in /home/vsts/work/1/s\repo/Release/test
Added Release/test/myNewWikiPage.md to repo /home/vsts/work/1/s\repo
Using the file - /home/vsts/work/1/s\repo/Release/test/.order
Creating a new .order file
Preppending entry to the .order file
##[error]Error: fatal: pathspec '/home/vsts/work/1/s\repo/Release/test/.order' did not match any files

##[error]Error: fatal: pathspec '/home/vsts/work/1/s\repo/Release/test/.order' did not match any files

Finishing: WikiUpdaterTask
```

The error occurs during [adding of the new .order file](https://github.com/rfennell/AzurePipelines/pull/1303/files#diff-24491c1a51dff38b84ec4b9191647dce228837741a2c15ebe6760a2718caaba4R267), but I couldn't figure out *why* the error occurs.

Ok, I have a repo of the problem. The problem only occurs if 

- The agent is running Linux
- The folder you are creating the .order file in does not exist

I suspect the issue is related to the `\` in 

> ##[error]Error: fatal: pathspec '/home/vsts/work/1/s\azurerepowithcreds/TestFolder1/.order' did not match any files

Working on a fix

--- a/Extensions/WikiUpdater/WikiUpdaterTask/src/GitWikiFuntions.ts
+++ b/Extensions/WikiUpdater/WikiUpdaterTask/src/GitWikiFuntions.ts
@@ -249,8 +249,11 @@ export async function UpdateGitWikiFile(
                     // as we are pre-pending we alway need a line feed
                     fs.writeFileSync(orderFile, `${entry}\r\n`);
                     fs.appendFileSync(orderFile, oldContent);
+                    logInfo(`Preppending entry to the .order file`);
+                } else {
+                    fs.writeFileSync(orderFile, `${entry}`);
+                    logInfo(`Creating .order file as it does not exist`);
                 }
-                logInfo(`Preppending entry to the .order file`);
             } else {
                 // appending the entry
                 if (fs.existsSync(orderFile)) {

